File: src/Kattis/COMP321/A4/B.py
while True:
    try:
        l = list(map(int, input().split()))
    except:
        break
    rocks = l[0]
    nm = l[1]
    moves = l[2:]
    l = [False] * (rocks + 1)
    for i in range(1, rocks + 1):
        for move in moves:
            if i - move >= 0 and not l[i - move]:
                l[i] = True
                break
            
    print("Stan wins" if l[rocks] else "Ollie wins")

File: src/Kattis/COMP321/A4/B.cpp
#include <iostream>

using namespace std;

int main()
{
    // Declare variables
    int num_rocks, num_moves, move;

    // Get input
    while (cin >> num_rocks >> num_moves)
    {
        // Get moves
        int moves[num_moves];
        for (int i = 0; i < num_moves; i++)
        {
            cin >> moves[i];
        }

        // Initialize boolean array
        bool win[num_rocks + 1];
        for (int i = 0; i < num_rocks + 1; i++)
        {
            win[i] = false;
        }

        // Populate array
        for (int i = 1; i < num_rocks + 1; i++)
        {
            for (int j = 0; j < num_moves; j++)
            {
                move = moves[j];
                if ((i - move >= 0) && !win[i - move])
                {
                    win[i] = true;
                    break;
                }
            }
        }

        // Output result
        if (win[num_rocks])
        {
            cout << "Stan wins" << endl;
        }
        else
        {
            cout << "Ollie wins" << endl;
        }
    }

    // Successful return
    return 0;
}
